### PR TITLE
Enable pc relative

### PR DESCRIPTION
**Pull request by luyahan**

### CODE REVIEW

File: src/codegen/riscv64/assembler-riscv64.cc [hunk 1]
```diff
@@ -686,23 +694,41 @@ int Assembler::CJumpOffset(Instr instr) {
   return imm12;
 }
 
-int Assembler::BrachlongOffset(Instr auipc, Instr instr_I) {
+int Assembler::BrachlongOffset(Instr auipc, Instr instr_I)  {
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64.cc:697]: Formatting (remove extra space)

File: src/codegen/riscv64/assembler-riscv64-inl.h [hunk 2]
```diff
@@ -179,6 +185,17 @@ Address RelocInfo::target_internal_reference_address() {
   return pc_;
 }
 
+Handle<Code> Assembler::relative_code_target_object_handle_at(
+    Address pc) const {
+  Instr instr1 = Assembler::instr_at(pc);
+  Instr instr2 = Assembler::instr_at(pc + kInstrSize);
+  DCHECK(IsAuipc(instr1));
+  DCHECK(IsJalr(instr2));
+  // Interpret 2 instructions generated by li (lui/ori).
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64-inl.h:194]: I don't understand this comment. Aren't the two instructions an `auipc` and a `jalr`?

File: src/codegen/riscv64/assembler-riscv64.cc [hunk 3]
```diff
@@ -661,7 +669,7 @@ bool Assembler::is_near_branch(Label* L) {
   return is_intn((pc_offset() - L->pos()), kBranchOffsetBits);
 }
 
-int Assembler::BranchOffset(Instr instr) {
+int Assembler::BranchOffset(Instr instr)  {
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64.cc:672]: Formatting (remove extra space)

File: src/codegen/riscv64/assembler-riscv64.cc [hunk 4]
```diff
@@ -2596,6 +2622,22 @@ int Assembler::RelocateInternalReference(RelocInfo::Mode rmode, Address pc,
   }
 }
 
+void Assembler::RelocateRelativeReference(RelocInfo::Mode rmode, Address pc,
+                                          intptr_t pc_delta) {
+  Instr instr = instr_at(pc);
+  Instr instr1 = instr_at(pc + 0 * kInstrSize);
```
Comment (fw-brice) [inline src/codegen/riscv64/assembler-riscv64.cc:2628]: `instr1` is the same as `instr`.

File: src/codegen/riscv64/macro-assembler-riscv64.cc [hunk 5]
```diff
@@ -2949,9 +2949,22 @@ void TurboAssembler::Jump(Handle<Code> code, RelocInfo::Mode rmode,
...
-      target_is_isolate_independent_builtin) {
+  if (target_is_isolate_independent_builtin &&
+      options().use_pc_relative_calls_and_jumps) {
+    int32_t code_target_index = AddCodeTarget(code);
+    Label skip;
+    BlockTrampolinePoolScope block_trampoline_pool(this);
+    if (cond != al) {
+      // By using delay slot, we always execute first instruction of
+      // GenPcRelativeJump (which is or_(t8, ra, zero_reg)).
+      Branch(&skip, NegateCondition(cond), rs, rt);
```
Comment (fw-brice) [inline src/codegen/riscv64/macro-assembler-riscv64.cc:2960]: We don't have a delay slot in RISC-V. I think this was copied from MIPS and should be modified.

File: src/codegen/riscv64/macro-assembler-riscv64.cc [hunk 6]
```diff
@@ -3017,8 +3030,24 @@ void TurboAssembler::Call(Handle<Code> code, RelocInfo::Mode rmode,
...
+  if (target_is_isolate_independent_builtin &&
+      options().use_pc_relative_calls_and_jumps) {
+    int32_t code_target_index = AddCodeTarget(code);
+    Label skip;
+    BlockTrampolinePoolScope block_trampoline_pool(this);
+    RecordCommentForOffHeapTrampoline(builtin_index);
+    if (cond != al) {
+      // By using delay slot, we always execute first instruction of
+      // GenPcRelativeJump (which is or_(t8, ra, zero_reg)).
+      Branch(&skip, NegateCondition(cond), rs, rt);
```
Comment (fw-brice) [inline src/codegen/riscv64/macro-assembler-riscv64.cc:3042]: Same as above regarding delay slot.

**Review by fw-brice — CHANGES_REQUESTED** [review]

File: src/codegen/riscv64/assembler-riscv64.cc (hunk 1 above)
Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64.cc]: done

File: src/codegen/riscv64/assembler-riscv64-inl.h (hunk 2 above)
Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64-inl.h]: done

File: src/codegen/riscv64/assembler-riscv64.cc (hunk 3 above)
Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64.cc]: done

File: src/codegen/riscv64/assembler-riscv64.cc (hunk 4 above)
Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64.cc]: had fixed it

File: src/codegen/riscv64/macro-assembler-riscv64.cc (hunk 5 above)
Comment (luyahan) [reply src/codegen/riscv64/macro-assembler-riscv64.cc:2960]: done

File: src/codegen/riscv64/macro-assembler-riscv64.cc (hunk 6 above)
Comment (luyahan) [reply src/codegen/riscv64/macro-assembler-riscv64.cc:3040]: done

**Review by qjivy — APPROVED** [review]

File: src/codegen/riscv64/constants-riscv64.h [hunk 7]
```diff
@@ -56,7 +56,7 @@ namespace v8 {
 namespace internal {
 
 // TODO(sigurds): Change this value once we use relative jumps.
```
Comment (fw-brice) [inline src/codegen/riscv64/constants-riscv64.h:58]: I think we can remove this TODO now.

**Review by fw-brice — CHANGES_REQUESTED** [review]
Just one new minor comment. Otherwise, it looks good to me.

**Review by fw-brice — APPROVED** [review]

**Pull request merged**